Lesson 1 Home Task for mentor review

vhoncharenko opened the pull request:
1. Не успел со стилями и дизайном вообще. Попрошу завтра знакомых дизайнеров сделать наброски нового внешнего вида. Пока что использовал то, что было.
2. Думаю, что где-то намутил  с БЭМ-нотацией, использую ее впервые и местами со code style.
3. И вообще это проект на node.js у меня первый в жизни. Не суди строго, пожалуйста. :-)

vhoncharenko committed:
Lesson 1 Home Task for mentor review

mykola-shyshov commented on client_src/test.js:
```diff
@@ -0,0 +1,32 @@
+        var buffer = [],
+            i, len;
+
+        for (i = 0, len = rows.length; i < len; ++i) {
```
.map(functiion

mykola-shyshov commented on client_src/test.js:
```diff
@@ -0,0 +1,32 @@
+
+    Handlebars.registerPartial( 'post-json', $( '#post-json-template' ).html() );
+
+    Handlebars.registerHelper( 'json', function(rows, options) {
```
split into json and table

mykola-shyshov commented:
add .gitignore

mykola-shyshov commented on package.json:
```diff
@@ -0,0 +1,28 @@
+  "description": "Frontend Advanced Course",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
```
"start" script

vhoncharenko committed:
Fixes after mentor review

Fixed:
1) Split helpers {{#json}} and {{#stripe}}
2) Fixed package.json - added start script
3) Fixes in code style
4) Fixed stripe helper
5) Added .gitignore in this commit

vhoncharenko committed:
Fixed test.html

mykola-shyshov commented on .gitignore:
```diff
@@ -0,0 +1 @@
+node_modules/
```
где  libs? руками не добавляешь? добавь сюда `**/libs`

mykola-shyshov commented on client_src/test.html:
```diff
@@ -0,0 +1,37 @@
+</div>
+
+<script type="text/x-handlebars-template" id="posts-json-template">
+    {{#json posts}}
```
json не блочный helper)

mykola-shyshov commented on client_src/test.js:
```diff
@@ -0,0 +1,52 @@
+    Handlebars.registerHelper('stripes', function(posts, options) {
+        return posts.map(function( post, index ) {
+            var postClass = (index % 2 == 0)
+                ? 'posts-container__stripes__post_even'
```
по красоте конечно было бы сюда передавать классы из вне.
П-р ис-я: 

```
{{stripes even=posts-container__stripes__post_even odd=...odd}}

тогда в хелпере
? options.hash.even : options.hash.odd

```

mykola-shyshov replied:
так по БЕМУ нельзя `posts-container__stripes__post_even`
класс так формирутеся - `Блок__Елемент_Ммодификатор`. 2 раза нельзя `__`

vhoncharenko committed:
Fixes after mentor review
Fixed:
1) Add **/libs to .gitignore
2) Fixed {{#json}} -> {{json}} (it's not a block helper)
3) Extended stripe helper functionality
4) Fixed name odd/even BEM-style class
